src/core/MOM_porous_barriers.F90

```diff
@@ -57,69 +58,74 @@ subroutine por_widths(h, tv, G, GV, US, eta, pbv, eta_bt, halo_size, eta_to_m)
-                 else
-                   pbv%por_face_areaU(I,j,k) = 0.0; endif
-              endif
-               eta_prev = eta_s
-               A_layer_prev = A_layer
-           endif; enddo
-       endif; enddo; enddo
+            else
+              pbv%por_face_areaU(I,j,k) = 0.0; endif
+          endif
+          eta_prev = eta_s
+          A_layer_prev = A_layer
+        endif
+      enddo
+    endif
+  enddo; enddo
 
   do i=isd,ied; do J=JsdB,JedB
-     if (G%porous_DavgV(i,J) < 0.) then
-        do K = nk+1,1,-1
-           eta_s = max(US%Z_to_m*eta(i,J,K), US%Z_to_m*eta(i,J+1,K)) !take shallower layer height
-           !eta_s = 0.5 * (US%Z_to_m*eta(i,J,K) + US%Z_to_m*eta(i,J+1,K)) !take arithmetic mean
-           if (eta_s <= G%porous_DminV(i,J)) then
-              pbv%por_layer_widthV(i,J,K) = 0.0
-              A_layer_prev = 0.0
-              if (K < nk+1) then
-                 pbv%por_face_areaV(i,J,k) = 0.0; endif
-           else
-              call calc_por_layer(G%porous_DminV(i,J), G%porous_DmaxV(i,J), G%porous_DavgV(i,J),&
-                   eta_s, w_layer, A_layer)
-              pbv%por_layer_widthV(i,J,K) = w_layer
-              if (k <= nk) then
-                 if ((eta_s - eta_prev) > 0.0) then
-                   pbv%por_face_areaV(i,J,k) = (A_layer - A_layer_prev)/&
+    if (G%porous_DavgV(i,J) < 0.) then
+      do K = nk+1,1,-1
+        eta_s = max(US%Z_to_m*eta(i,J,K), US%Z_to_m*eta(i,J+1,K)) !take shallower layer height
+        !eta_s = 0.5 * (US%Z_to_m*eta(i,J,K) + US%Z_to_m*eta(i,J+1,K)) !take arithmetic mean
+        if (eta_s <= G%porous_DminV(i,J)) then
+          pbv%por_layer_widthV(i,J,K) = 0.0
+          A_layer_prev = 0.0
+          if (K < nk+1) then
+            pbv%por_face_areaV(i,J,k) = 0.0; endif
+        else
+          call calc_por_layer(US%Z_to_m*(G%porous_DminV(i,J)-G%Z_ref), &
+            US%Z_to_m*(G%porous_DmaxV(i,J)-G%Z_ref), &
+            US%Z_to_m*(G%porous_DavgV(i,J)-G%Z_ref), eta_s, w_layer, A_layer)
+          pbv%por_layer_widthV(i,J,K) = w_layer
+          if (k <= nk) then
+            if ((eta_s - eta_prev) > 0.0) then
+              pbv%por_face_areaV(i,J,k) = (A_layer - A_layer_prev)/&
                    (eta_s-eta_prev)
-                 else
-                   pbv%por_face_areaU(I,j,k) = 0.0; endif
-              endif
-               eta_prev = eta_s
-               A_layer_prev = A_layer
-           endif; enddo
-       endif; enddo; enddo
+            else
+              pbv%por_face_areaU(I,j,k) = 0.0; endif
+          endif
+          eta_prev = eta_s
+          A_layer_prev = A_layer
+        endif
+      enddo
+    endif
+  enddo; enddo
 
 end subroutine por_widths
 
+!> subroutine to calculate the profile fit for a single layer in a column
 subroutine calc_por_layer(D_min, D_max, D_avg, eta_layer, w_layer, A_layer)
-!subroutine to calculate the profile fit for a layer
 
   real,            intent(in) :: D_min !< minimum topographic height [m]
```

Comment:
The 4 arguments D_min, D_max, D_avg, and eta_layer to calc_por_layer() should also be in units of [Z ~> m] and should all use the same offset G%Z_ref, but because of they way they are used in this routine, it is only the comments describing the variables that need to change.

Also, as coded here, A_layer has units of [m] (or perhaps it will become [Z]), and not [nondim] as documented in the comment.